sources/rocky-http.go

```diff
@@ -0,0 +1,524 @@
+package sources
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	lxd "github.com/lxc/lxd/shared"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+
+	"github.com/lxc/distrobuilder/shared"
+)
+
+// RockyOSHTTP represents the Rocky HTTP downloader.
+type  struct {
```

Comment:
this struct doesn't have a name

Reply:
OK, added struct name.

Reply:
We've managed to totally munge the rocky-http.go file to get an image, but not in a way that uses the variables so it can be re-used later. Is there, anywhere, a guide on what is minimally required for a working .go file for this?

Reply:
@sspencerwire You can look at : https://github.com/AlmaLinux/distrobuilder/blob/ed19bc4a74b79ef6e52df873905bce6b3f626262/sources/almalinux-http.go
For example or CentOS 8.